posthog/temporal/tests/batch_exports/test_monitoring.py

```diff
@@ -187,11 +196,30 @@ async def test_monitoring_workflow(
             expected_missing_runs.append((run.data_interval_start, run.data_interval_end))
             await run.adelete()
 
+    if simulate_zero_records_completed:
```

Comment:
There is a tradeoff between DRY and complexity, and I think with this we are certainly making things more complex.

I don't necessarily think we are crossing any thresholds, but I would start looking into splitting up this test in follow-up work.